### ISSUE TITLE
Conform confound names with draft derivatives spec

### ISSUE THREAD

**Issue by effigies** [poldracklab/fmriprep#1864]
Current confounds include suffixes like `_derivative1` and `_power2`. The [draft Functional Derivatives spec](https://bids-specification.readthedocs.io/en/derivatives/05-derivatives/04-functional-derivatives.html#time-series-transformations) would use `_dt` and `_sq`.

**Comment by jdkent** [comment]
- [Here is where power is used](https://github.com/poldracklab/niworkflows/blob/bf96c7c4cca990b31f9d496f19372307721fee0b/niworkflows/interfaces/confounds.py#L321)
- [Here is where derivative is used](https://github.com/poldracklab/niworkflows/blob/bf96c7c4cca990b31f9d496f19372307721fee0b/niworkflows/interfaces/confounds.py#L277)

As suggested by the above variable names, the code was written to be flexible to account for arbitrary order derivatives and powers, not just the special cases of 1st order derivatives and 2nd order powers.

So at this point the relevant code change would be to make first order derivative and second order power special cases and keep the code flexible to include the unspecified time series transformation. Or should we just remove the possibility of doing anything other than 1st order derivatives and 2nd order powers? (I can do this if we agree on the path to take).

Would it be worth it to open a discussion on adopting a more flexible naming scheme for these transformations in functional derivatives?

**Comment by effigies** [comment]
Yeah, I guess it's worth opening up the conversation about what to do with arbitrary transformations of time series. If people are clamoring for Taylor expansions, there's no point in changing the names now only to change them back again later...

Personally, I don't see a large benefit in performing these computations in fMRIPrep. Given that they are deterministic transformations of the columns we already provide, consumers can perform these operations as needed.

But I also didn't pay attention to any arguments that may have been had when this was added, so perhaps this has already been justified. @rciric @oesteban?
